tests/backend/cores/abc_test_dbcore.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+# Import standard library
+import abc
+
+# Import modules
+import pytest
+from sqlalchemy.engine.base import Engine
+from sqlalchemy.schema import Table
+
+
+class ABCTestDBCore(abc.ABC):
+    """Base Test class for all DBCore implementations"""
+
+    @pytest.fixture
+    def core(self):
+        raise NotImplementedError
+
+    @pytest.fixture
+    def config(self):
+        raise NotImplementedError
+
+    @pytest.fixture
+    def test_tables(self):
+        raise NotImplementedError
+
+    @pytest.mark.usefixtures("core")
+    def test_get_engine_return_type(self, core):
+        """Test if get_engine() returns appropriate type"""
+        engine = core.get_engine()
+        assert isinstance(engine, Engine)
+
+    @pytest.mark.usefixtures("core", "config", "sample_points", "test_tables")
+    def test_get_tables_return_type(
```

Comment:
Not a useful test? Better checks IMO:
- Source table name is same same as input?
- Target table name is of uuid format?
- Columns are as expected?

Reply:
Ahh that makes sense. Sige will add those checks later!